### DISCUSSION TITLE
When inserting lof and lot in toc, the page number is not displayed correctly.

### DISCUSSION THREAD

**Discussion by MasterGPT9821** [quarto-dev/quarto-cli#8440] [Q&A]
### Description

How to display the page number correctly when inserting lof and lot in toc.

My qmd
```qmd
---
title: "Titel"
subtitle: "Untertitel"
author:
  - "Max Mustermann"
  - "Maria Musterfrau"
date: today
lang: de
format:
  pdf:
    documentclass: scrbook
    keep-tex: true
    toc: true
    toc-depth: 3
    lof: true
    lot: true
    number-sections: true
    include-in-header:
      - header_footer.tex
fontsize: 11pt
linestretch: 1.5
geometry: left = 25mm, right = 25mm, top = 30mm, bottom = 30mm
bibliography: bosch.bib
execute: 
  echo: false
  warning: false
  message: false
editor: visual
---

```latex
\addcontentsline{toc}{chapter}{\listfigurename} % Füge das Verzeichnis der Abbildungen zum Inhaltsverzeichnis hinzu
\addcontentsline{toc}{chapter}{\listtablename} % Füge das Verzeichnis der Abbildungen zum Inhaltsverzeichnis hinzu
```

Here the tex-code separatly for a better view:
```latex
\addcontentsline{toc}{chapter}{\listfigurename} % Füge das Verzeichnis der Abbildungen zum Inhaltsverzeichnis hinzu
\addcontentsline{toc}{chapter}{\listtablename} % Füge das Verzeichnis der Abbildungen zum Inhaltsverzeichnis hinzu
```

the output
![grafik](https://github.com/quarto-dev/quarto-cli/assets/157615015/e3573b1b-c5e0-426e-ba5e-109b5f40a255)
![grafik](https://github.com/quarto-dev/quarto-cli/assets/157615015/9e25e0a6-211b-4734-a83e-ba3b68568ddc)

**Comment by cderv** [comment]
Can you share your whole qmd file ? Or a smaller one that reproduce the issue ? 
thanks

**Reply by MasterGPT9821** [reply]
[example_quarto_book_individual_title.txt](https://github.com/quarto-dev/quarto-cli/files/14083804/example_quarto_book_individual_title.txt)
i save the qmd as txt so that you can copy it. Do you need the other files? Should i make a repository

**Reply by MasterGPT9821** [reply]
a part of the other files
![logo](https://github.com/quarto-dev/quarto-cli/assets/157615015/23f446c5-d687-4084-a47d-f0f5c6576290)
![logo](https://github.com/quarto-dev/quarto-cli/assets/157615015/05a186be-5370-4f8d-8cf1-14191c194617)
[test.txt](https://github.com/quarto-dev/quarto-cli/files/14083841/test.txt)
[_titlepage.txt](https://github.com/quarto-dev/quarto-cli/files/14083843/_titlepage.txt)
[before-body.txt](https://github.com/quarto-dev/quarto-cli/files/14083844/before-body.txt)

**Reply by cderv** [reply]
When there are several files, it is best to create a GitHub repo or upload a zip file. This makes it easier for us to get your example and reproduce it. 

Try to simplify your document to test that 
```latex
\addcontentsline{toc}{chapter}{\listfigurename} % Füge das Verzeichnis der Abbildungen zum Inhaltsverzeichnis hinzu
\addcontentsline{toc}{chapter}{\listtablename} % Füge das Verzeichnis der Abbildungen zum Inhaltsverzeichnis hinzu
```

is indeed working in a simple .pdf document format with Quarto. And if it is working, add one by one your customization to see why it could break. 

As I said, this does not seem like a Quarto issue, but something with how to do that with LaTeX. 

Thanks a lot for helping us help you.

**Comment by cderv** [comment]
Also if this can help, interesting resource with an automatic solution using `tocbibind` package: https://www.ctan.org/pkg/tocbibind

https://texblog.org/2013/04/29/latex-table-of-contents-list-of-figurestables-and-some-customizations/

**Reply by MasterGPT9821** [reply]
so i have to insert this package before the document begins?

**Reply by cderv** [reply]
You can use [latex includes](https://quarto.org/docs/output-formats/pdf-basics.html#latex-includes) in YAML to add content. It could solve your issue. Yes I did not test it in your case because your example is incomplete - as there is no content, there is not TOC created, nor figures or tables to list. 

Provide a full reproducible example if you want more help on how to do things.

I believe this is a LaTeX issue here - in the sense that Quarto does not do that numbering, and adding list of tables and figures is done using LaTeX code directly.

**Reply by MasterGPT9821** [reply]
with before-body-tex ?

**Reply by cderv** [reply]
It depends where this should be inserted. This probably mentioned in example about this package. Did you try something ? 

Test with simple document always to understand how things works, before applying to your main document often already with way more customization that could conflict.

**Reply by MasterGPT9821** [reply]
my document is very simple an not the main document...

**Comment by MasterGPT9821** [comment]
I uploaded the files to a new repository:
https://github.com/MasterGPT9821/Single-qmd-file-example

**Reply by MasterGPT9821** [reply]
@cderv does this help?

**Reply by cderv** [reply]
I can't run the example. There are some error by cloning and running `quarto render` with latest 1.4 version. 

Can you try a simpler example as I adviced so that you check that the LaTeX code you are including is ok. Adding this type of content 
````
\addcontentsline{toc}{chapter}{\listfigurename} % Füge das Verzeichnis der Abbildungen zum Inhaltsverzeichnis hinzu
\addcontentsline{toc}{chapter}{\listtablename} % Füge das Verzeichnis der Abbildungen zum Inhaltsverzeichnis hinzu

````

is a LaTeX way to achieve your aim, and not a Quarto way. So if it does not work without any Quarto error, then you need to find what is wrong in the LaTeX file produced (`keep-tex: true)` to see it. 

If Quarto is involved in why the LaTeX produced is not working, please do report that by sharing simple reproducible example - one that works for LaTeX only maybe, and another with simple quarto doc that does not work, and what is the difference. 

THanks !

**Reply by MasterGPT9821** [reply]
i added the .tex file in the repository

**Reply by cderv** [reply]
@MasterGPT9821 I can help only if you provide something that will help me help you. 

There is still issue by 
- cloning your repo 
- running `quarto render .\example_quarto_book_individual_title.qmd`

```
ERROR: Template partial template/before-body.tex was not found
```

I adviced you several time to create a smaller example, remove every customization except the lot / lof in TOC thing. You did not yet follow that. 

You should really spend time on creating this example, this will probably unblock you by understanding how this all work. Right now, example is unusable. 

Another advice: To test a reproducible example, you need to put you in the position of the user that will use your example. So I don't have your setup. You then need to be sure 
- to test on a clean project environment (like temp folder). 
- Using same step I would use (cloning before running). 

Anyhow, please to provide what help us help you, otherwise we can't do much

**Reply by MasterGPT9821** [reply]
I dont't understand the ERROR, because when i render the qmd there is no error

**Reply by cderv** [reply]
Probably file was missing, and you just added it in https://github.com/MasterGPT9821/Single-qmd-file-example/commit/c87949b770c93201beb8263ca725c883f5a6298a

**Reply by MasterGPT9821** [reply]
I'm glad it's working for you now too.

**Comment by MasterGPT9821** [comment]
i fixed the problem:

This is of no use:
```
\addcontentsline{toc}{chapter}{\listfigurename} % Füge das Verzeichnis der Abbildungen zum Inhaltsverzeichnis hinzu
\addcontentsline{toc}{chapter}{\listtablename} % Füge das Verzeichnis der Abbildungen zum Inhaltsverzeichnis hinzu
```

instead it is necessary to insert `\usepackage[nottoc]{tocbibind}` in the template `(template.tex).`
I did by including the expression `\usepackage[nottoc]{tocbibind}` into the default `doc-class.tex` (https://github.com/quarto-dev/quarto-cli/blob/main/src/resources/formats/pdf/pandoc/doc-class.tex).

My `template-partial` `doc-class.tex`:
```latex
\documentclass[
$if(fontsize)$
  $fontsize$,
$endif$
$if(papersize)$
  $papersize$paper,
$endif$
$if(beamer)$
  ignorenonframetext,
$if(handout)$
  handout,
$endif$
$if(aspectratio)$
  aspectratio=$aspectratio$,
$endif$
$endif$
$for(classoption)$
  $classoption$$sep$,
$endfor$
]{$documentclass$}

% ##############################################
% ### default above, addet expression below ###
% ##############################################

\usepackage[nottoc]{tocbibind}  % Package to include lists in ToC, excluding ToC itself
```

My YAML is now:
```qmd
---
title: "Titel"
subtitle: "Untertitel"
author:
  - "Max Mustermann"
  - "Maria Musterfrau"
date: today
lang: de
format:
  pdf:
    documentclass: scrbook
    keep-tex: true
    toc: true
    toc-depth: 3
    lof: true
    lot: true
    number-sections: true
    include-in-header:
      - "header_footer.tex"
    template-partials:
      - "template/doc-class.tex"
      - "template/before-body.tex"
      - "template/_titlepage.tex"
fontsize: 11pt
linestretch: 1.5
geometry: left = 25mm, right = 25mm, top = 30mm, bottom = 30mm
bibliography: bib.bib
execute: 
  echo: false
  warning: false
  message: false
editor: visual
---
```

i also updated the repository: https://github.com/MasterGPT9821/Single-qmd-file-example

**Reply by MasterGPT9821** [reply]
as you can see in the keeping tex-file: `\usepackage[nottoc]{tocbibind}` was taken over
![grafik](https://github.com/quarto-dev/quarto-cli/assets/157615015/f6a543ef-a122-4d83-944c-787492fb8b04)

**Reply by cderv** [reply]
> i fixed the problem:

Awesome ! Glad you finally found how to make this work. 


You should probably be able to make this works inserting `tocbibind` package call using [Latex includes](https://quarto.org/docs/output-formats/pdf-basics.html#latex-includes)

Modifying the partial or the template is usually for more advanced customization where you need to play with Pandoc's variable. 

Anyhow, this works so that is the most important !
